pandas/core/reshape/concat.py

```diff
@@ -219,6 +220,11 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
         ...
     ValueError: Indexes have overlapping values: ['a']
     """
+    if join_axes is not None:
```

Comment:
you would need to remove its use inside the function.

further this is called in several places in the code base they would need to be changed. Are current tests showing a deprecation warning?

Reply:
> you would need to remove its use inside the function.

Um, isn't there a deprecation cycle between issuing the warning and removing the keyword?

> further this is called in several places in the code base they would need to be changed. Are current tests showing a deprecation warning?

It was called in 3 places. Changed those, now there shouldn't be any warnings anymore (will check against log of current travis run)

Reply:
of course there is a deprecation cycle. I am not talking about the public API, rather the private usage of this.

Reply:
@jreback Then what do you want me to remove? I don't know all the possible uses that someone might have used `join_axes`, (e.g. replacing the call in `pandas/core/generic.py` with `reindex` would change the order of the index and then fail some tests), so replacing it internally should wait until complete deprecation, no?

Reply:
if its a useless argument then let's actually remove it internally. If its really really hard then pls indicate that. But much better to completely remove the usage of it if possible.

Reply:
@jreback It's not useless per se, just redundant, obscure, and not well documented. I *could* replace its internal uses, but:
- I might silently break code
- the deprecation suggests to use `reindex` on the *result*, so changing something *within* the methdod is IMO pointless until it's ripped out completely.

Reply:
@jreback I still think that removing the internal uses might break things. Shouldn't the complete removal wait until 1.0?

Reply:
no if you are deprecating a public function, then we want to remove the implemenation now and its uses internally.